tests/unit/charms/tls_certificates_interface/v2/test_tls_certificates_v2_provides.py

```diff
@@ -185,6 +211,92 @@ def test_given_certificate_for_csr_already_in_relation_data_when_on_relation_cha
 
         patch_certificate_creation_request.assert_not_called()
 
+    @patch(
+        f"{LIB_DIR}.CertificatesProviderCharmEvents.certificate_creation_request",
+        new_callable=PropertyMock,
+    )
+    def test_given_certificate_for_csr_already_in_relation_data_when_on_relation_changed_and_unit_not_leader_then_certificate_creation_request_is_not_emitted(  # noqa: E501
```

Comment:
Should we keep this test now that the `is_leader` guard is added at the beginning of `on_config_changed`? I mean the cert creation request is not going to be emitted independently from wether the csr is in the databag or not.

Reply:
Here, certificate request is not emitted because unit is not leader. csr is in relation databag or not does not have any effect.This test is not beneficial anymore.

Reply:
Done